spec.md

```diff
@@ -1663,6 +1708,398 @@ message GetCapacityResponse {
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Opaque static properties of the volume group.
+  // This field is OPTIONAL.
+  map<string, string> volume_group_context = 2;
+
+  // Underlying volumes in this group. The same definition in CSI
+  // Volume.
+  // This field is REQUIRED.
+  // To support the creation of an empty group, this list can be empty.
+  // However, this field is not empty in the following cases:
+  // - Response from ListVolumeGroups or GetVolumeGroup if the
+  //   VolumeGroup is not empty.
+  // - Response from ModifyVolumeGroup if the VolumeGroup is not
+  //   empty after modification.
+  // - Phase 2: Create a new volume group from a source group snapshot.
+  // - Phase 2: Create a new volume group and add a list of existing
+  //   volumes to the group.
+  repeated .csi.v1.Volume volumes = 3;
+}
+```
+
+#### `CreateVolume`
+
+TODO
+
+#### `DeleteVolumeGroup`
+
+**ALPHA FEATURE**
+
+```protobuf
+message DeleteVolumeGroupRequest {
+  option (alpha_message) = true;
+
+  // The ID of the volume group to be deprovisioned.
+  // This field is REQUIRED.
+  string volume_group_id = 1;
+
+  // Secrets required by plugin to complete volume group deletion
+  // request.
+  // This field is OPTIONAL. Refer to the `Secrets Requirements`
+  // section on how to use this field.
+  map<string, string> secrets = 2 [(csi_secret) = true];
+}
+
+message DeleteVolumeGroupResponse {
+  option (alpha_message) = true;
+  // Intentionally empty.
+}
+```
+
+#### `ModifyVolumeGroup`
+
+**ALPHA FEATURE**
+
+This RPC will be called by the CO to modify an existing volumegroup on behalf of a user. volume_ids provided in the ModifyVolumeGroupRequest will be compared to the ones in the existing VolumeGroup. New volume_ids in the modified VolumeGroup will be added to the VolumeGroup. Existing volume_ids not in the modified VolumeGroup will be removed from the VolumeGroup. If volume_ids is empty, the VolumeGroup will be removed of all existing volumes. This operation MUST be idempotent.
+
+To support ModifyVolumeGroup, the Kubernetes VolumeGroup controller will be implemented to have a desired state of the world and an actual state of the world. The desired state of the world contains VolumeGroups with the desired PVCList while the actual state of the world contains VolumeGroups with the actual PVCList. The controller will try to reconcile the two by handling adding and removing multiple PVCs through a single CSI RPC call each time.
+
+Note that filesystems based storage systems may not be able to support this RPC. For block based storage systems, this is a very convenient method. However, it may not satisfy the requirement for consistency as the volume is created without the knowledge of which group it is placed in. It is out of the scope of the CSI spec to determine whether a group is consistent or not. It is up to the storage provider to clarify that in the vendor specific documentation.
```

Comment:
Great note about the consistency! It is important to be clear that VolumeGroups are not "Consistency Groups, but vendors may want to use their "consistency group" feature to implement VolumeGroups.

I think it would be good to describe possible error values here.

- a volume was added that is incompatible with other volumes in the group
- group can not add more volumes, too many volumes, or group too small, maximum volume reached

Reply:
Good point.  Added error codes.